packages/vuetify/src/util/helpers.ts

```diff
@@ -70,6 +70,8 @@ export function getObjectValueByPath (obj: any, path?: string | null, fallback?:
   return getNestedValue(obj, path.split('.'), fallback)
 }
 
+export type SelectItemId = string | undefined | ((item: any) => string)
```

Comment:
Add `number` to the function: `string | number`, we should review the type returned here, we also need the `item-id` prop, in any case should it should be a primivite.

We can keep `string`, the user should add it using proper conversion: `id: number | string`